src/core/unittest/CubicTest.cpp

```diff
@@ -0,0 +1,808 @@
+
+    CubicCongestionControlInitialize(&Connection.CongestionControl, &Settings);
+
+    QUIC_CONGESTION_CONTROL_CUBIC* Cubic = &Connection.CongestionControl.Cubic;
+
+    // Set BytesInFlight to half the window to have available capacity
+    Cubic->BytesInFlight = Cubic->CongestionWindow / 2;
+    uint32_t AvailableWindow = Cubic->CongestionWindow - Cubic->BytesInFlight;
+
+    // Simulate 10ms elapsed since last send
+    // Expected pacing calculation: (CongestionWindow * 10ms) / 50ms = CongestionWindow / 5
+    uint32_t TimeSinceLastSend = 10000; // 10ms in microseconds
+
+    uint32_t Allowance = Connection.CongestionControl.QuicCongestionControlGetSendAllowance(
+        &Connection.CongestionControl, TimeSinceLastSend, TRUE);
+
+    // Pacing should limit the allowance to less than the full available window
+    ASSERT_GT(Allowance, 0u); // Should allow some sending
+    ASSERT_LT(Allowance, AvailableWindow); // But less than full window due to pacing
+
+    // Verify it's approximately the expected pacing calculation
+    uint32_t ExpectedPacedAllowance = (uint32_t)(((uint64_t)Cubic->CongestionWindow * TimeSinceLastSend) / Connection.Paths[0].SmoothedRtt);
+
+    // Allow some margin due to integer arithmetic and min/max clamping
+    ASSERT_GE(Allowance, ExpectedPacedAllowance / 2);
+    ASSERT_LE(Allowance, ExpectedPacedAllowance * 2);
+}
+
+//
+// Test 8: Getter functions (via function pointers)
+// Scenario: Tests all simple getter functions that return internal state values.
+// Verifies GetExemptions, GetBytesInFlightMax, and GetCongestionWindow all return
+// correct values matching the internal CUBIC state.
+//
+TEST(CubicTest, GetterFunctions)
+{
+    QUIC_CONNECTION Connection;
+    QUIC_SETTINGS_INTERNAL Settings;
+    CxPlatZeroMemory(&Settings, sizeof(Settings));
+
+    Settings.InitialWindowPackets = 10;
+    Settings.SendIdleTimeoutMs = 1000;
+
+    InitializeMockConnection(&Connection, 1280);
+    CubicCongestionControlInitialize(&Connection.CongestionControl, &Settings);
+
+    QUIC_CONGESTION_CONTROL_CUBIC *Cubic = &Connection.CongestionControl.Cubic;
+
+    // Test GetExemptions
+    uint8_t Exemptions = Connection.CongestionControl.QuicCongestionControlGetExemptions(&Connection.CongestionControl);
+    ASSERT_EQ(Exemptions, 0u);
+    Cubic->Exemptions = 3;
+    Exemptions = Connection.CongestionControl.QuicCongestionControlGetExemptions(&Connection.CongestionControl);
+    ASSERT_EQ(Exemptions, 3u);
+
+    // Test GetBytesInFlightMax
+    uint32_t MaxBytes = Connection.CongestionControl.QuicCongestionControlGetBytesInFlightMax(&Connection.CongestionControl);
+    ASSERT_EQ(MaxBytes, Cubic->BytesInFlightMax);
+    ASSERT_EQ(MaxBytes, Cubic->CongestionWindow / 2);
+
+    // Test GetCongestionWindow
+    uint32_t CongestionWindow = Connection.CongestionControl.QuicCongestionControlGetCongestionWindow(&Connection.CongestionControl);
+    ASSERT_EQ(CongestionWindow, Cubic->CongestionWindow);
```

Comment:
nit: Some of these getter tests feel very low value - the assertion is basically the function implementation.